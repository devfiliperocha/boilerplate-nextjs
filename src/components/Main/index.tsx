import * as S from './styles' /** S = Styles */

const Main = ({
  title = 'Desenvolvimento de Software e Aplicativos Personalizados de Alta Performance',
  description = 'Uma idéia no papel, é só uma idéia...',
  logo = '/img/logo.svg',
  illustration = '/img/hero-illustration.svg'
}) => (
  <S.Wrapper>
    <main>
      <S.Logo
        src={logo}
        alt="Um Rosto com Capuz, simbolizando o logo, e o nome Phantom Software House ao lado"
      />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration
        src={illustration}
        alt="Desenvolvedor sentado de frente para telas com códigos"
      />
      <br />
    </main>
  </S.Wrapper>
)

export default Main
