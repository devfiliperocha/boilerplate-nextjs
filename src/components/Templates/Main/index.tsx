import * as S from './styles' /** S = Styles */

const Main = ({
  title = '',
  description = '',
  logo = '/img/logo.svg',
  illustration = '/img/hero-illustration.svg'
}) => (
  <S.Wrapper>
    <main>
      <S.Logo src={logo} alt="" />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration src={illustration} alt="" />
      <br />
    </main>
  </S.Wrapper>
)

export default Main
