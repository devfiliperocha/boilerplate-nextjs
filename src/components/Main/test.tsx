import { screen, render } from '@testing-library/react'
import Main from '.'

describe('<Main/>', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /Desenvolvimento de Software/i })
    ).toBeInTheDocument()
    // Cria um snapshot para comparar as mudanÃ§as nos arquivos com ele.
    // Para garantir
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#1b2936' })
  })
})
