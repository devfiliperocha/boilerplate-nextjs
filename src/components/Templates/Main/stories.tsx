import Main from '.'
import { Meta, Story } from '@storybook/react/types-6-0'

export default {
  title: 'Main',
  component: Main
} as Meta

export const Basic: Story = (args) => <Main {...args} />
Basic.argTypes = {
  title: {
    name: 'Title',
    type: { name: 'string' },
    defaultValue:
      'Desenvolvimento de Software e Aplicativos Personalizados de Alta Performance',
    description: 'Título',
    control: {
      type: 'text'
    }
  },
  description: {
    name: 'Description',
    type: { name: 'string' },
    defaultValue: 'Uma idéia no papel, é só uma idéia...',
    description: 'Descrição',
    control: {
      type: 'text'
    }
  },
  logo: {
    name: 'Logo',
    type: { name: 'string' },
    description: 'Arquivo de logo',
    control: {
      type: 'file'
    }
  },
  illustration: {
    name: 'Illustration',
    type: { name: 'string' },
    description: 'Arquivo de ilustração',
    control: {
      type: 'file'
    }
  }
}
