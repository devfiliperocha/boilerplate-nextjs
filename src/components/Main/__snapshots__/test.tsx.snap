// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render the heading 1`] = `
.c0 {
  background-color: #1b2936;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  height: 5.5rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2rem;
  width: 45rem;
  line-height: 2rem;
  margin: 2rem;
}

.c3 {
  font-size: 1.6rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  font-weight: 200;
  color: lightgrey;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
  height: min(30rem,100%);
}

<main
  class="c0"
>
  <main>
    <img
      alt="Um Rosto com Capuz, simbolizando o logo, e o nome Phantom Software House ao lado"
      class="c1"
      src="/img/logo.svg"
    />
    <h1
      class="c2"
    >
      Desenvolvimento de Software e Aplicativos Personalizados de Alta Performance
    </h1>
    <h2
      class="c3"
    >
      Uma idéia no papel, é só uma idéia...
    </h2>
    <img
      alt="Desenvolvedor sentado de frente para telas com códigos"
      class="c4"
      src="/img/hero-illustration.svg"
    />
    <br />
  </main>
</main>
`
